shared/AppInsightsCommon/Tests/Unit/src/AppInsightsCommon.tests.ts

```diff
@@ -293,5 +293,116 @@ export class ApplicationInsightsTests extends AITestClass {
                 loggerStub.restore();
             }
         });
+        this.testCase({
+            name: 'DataSanitizerTests: dataSanitizeUrl properly redacts credentials in URLs with config enabled',
+            test: () => {
+                // URLs with credentials
+                let config = {} as IConfiguration;
+                const urlWithCredentials = "https://username:password@example.com/path";
+                const expectedRedactedUrl = "https://REDACTED:REDACTED@example.com/path";
+
+                // Act & Assert
+                const result = dataSanitizeUrl(this.logger, urlWithCredentials, config);
+                Assert.equal(expectedRedactedUrl, result);
+            }
+        });
+        this.testCase({
+            name: 'DataSanitizerTests: dataSanitizeUrl properly redacts credentials in URLs with config disabled',
+            test: () => {
+                // URLs with credentials
+                let config = {redactUrls: false} as IConfiguration;
+                const urlWithCredentials = "https://username:password@example.com/path";
```

Comment:
Lets add some additional tests where the username / password has encoded values or even if they are invalid, like someone tries to pass in an invalid url with the same url (for some reason)

eg something unexpected like, but it starts to "look" like a username / password
`https://file:///C/user....`